README.md

```diff
@@ -9,41 +9,52 @@
-determine the semantic purpose of the functions in a parser. It also has an experimental feature capable of generating a
-context free grammar representing the language accepted by a parser.
-
-Unlike dynamic instrumentation alternatives like [Taintgrind](https://github.com/wmkhoo/taintgrind), PolyTracker imposes
-negligible performance overhead for almost all inputs, and is capable of tracking every byte of input at once.
-PolyTracker started as a fork of the LLVM DataFlowSanitizer and takes much inspiration from the
-[Angora Fuzzer](https://github.com/AngoraFuzzer/Angora). However, unlike the Angora system, PolyTracker is able to track
-the entire _provenance_ of a taint. In February of 2021, the LLVM DataFlowSanitizer added a new feature for tracking
-taint provenance called [_origin tracking_](https://reviews.llvm.org/D95835). However, it is only able to track at most
-16 taints at once, while PolyTracker can track up to 2³².
-
-This README serves as the general usage guide for installing PolyTracker and compiling/instrumenting binaries. For
-programmatically interacting with or extending PolyTracker through its Python API, as well as for interacting with
+PolyTracker is a tool originally created for the _Automated Lexical Annotation
+and Navigation of Parsers_, a backronym devised solely for the purpose of
+referring to it as _The ALAN Parsers Project_. However, it has evolved into a
+general purpose tool for efficiently performing data-flow and control-flow
+analysis of programs. PolyTracker is an LLVM pass that instruments programs to
+track which bytes of an input file are operated on by which functions. It
+outputs a database containing the data-flow information, as well as a runtime
+trace. PolyTracker also provides a Python library for interacting with and
+analyzing its output, as well as an interactive Python REPL.
+
+PolyTracker can be used in conjunction with
+[PolyFile](https://github.com/trailofbits/polyfile) to automatically determine
+the semantic purpose of the functions in a parser. It also has an experimental
+feature capable of generating a context free grammar representing the language
+accepted by a parser.
+
+Unlike dynamic instrumentation alternatives like
+[Taintgrind](https://github.com/wmkhoo/taintgrind), PolyTracker imposes
+negligible performance overhead for almost all inputs, and is capable of
+tracking every byte of input at once. PolyTracker started as a fork of the LLVM
+DataFlowSanitizer and takes much inspiration from the
+[Angora Fuzzer](https://github.com/AngoraFuzzer/Angora). However, unlike the
+Angora system, PolyTracker is able to track the entire _provenance_ of a taint.
+In February of 2021, the LLVM DataFlowSanitizer added a new feature for tracking
+taint provenance called [_origin tracking_](https://reviews.llvm.org/D95835).
+However, it is only able to track at most 16 taints at once, while PolyTracker
+can track up to 2³².
```

Comment:
This is actually not entirely true...
We can generate 2^31-1 labels in total (label zero is reserved for non-tainted).